Packages/com.unity.input/Runtime/Input.cs

```diff
@@ -647,27 +648,125 @@ public static bool GetKeyUp(KeyCode key)
 
-        #region Unimplemented
+        private static TouchPhase ToLegacy(InputSystem.TouchPhase phase)
+        {
+            switch (phase)
+            {
+                case InputSystem.TouchPhase.Began: return TouchPhase.Began;
+                case InputSystem.TouchPhase.Moved: return TouchPhase.Moved;
+                case InputSystem.TouchPhase.Ended: return TouchPhase.Ended;
+                case InputSystem.TouchPhase.Canceled: return TouchPhase.Canceled;
+                case InputSystem.TouchPhase.Stationary: return TouchPhase.Stationary;
+                    //InputSystem.TouchPhase.None  // These are filtered out and shouldn't be seen
+            }
+            throw new ArgumentOutOfRangeException(nameof(phase), $"Not expected phase value: {phase}");
+        }
 
-        public static float GetAxisRaw(string axisName)
+        private static (float radius, float variance) RadiusValues(Vector2 vRadius)
         {
-            ////TODO: implement properly
-            return GetAxis(axisName);
+            var x = vRadius.x;
+            var y = vRadius.y;
+            var max = Mathf.Max(x, y);
+            var min = Mathf.Min(x, y);
+            var variance = (max - min) / 2;
+            var radius = min + variance;
+
+            return (radius, variance);
+        }
+
+        private static (float altitudeAngle, float azimuthAngle) TiltValues(Vector2 vTilt)
+        {
+            var azimuthAngle = Mathf.Acos(vTilt.x);
+            var altitudeAngle = Mathf.Acos(vTilt.y);
+            return (altitudeAngle, azimuthAngle);
+        }
+
+        public static Touch ToTouch(InputSystem.Controls.TouchControl control)
+        {
+            Touch touch = new Touch();
+            touch.fingerId = control.touchId.ReadValue();
+            touch.position = control.position.ReadValue();
+            touch.rawPosition = control.startPosition.ReadValue();
+            touch.deltaPosition = control.delta.ReadValue();
+            //touch.deltaTime = TODO: How to implement this? Device has lastUpdateTime() but only for single touch
+            touch.tapCount = control.tapCount.ReadValue();
+            touch.phase = ToLegacy(control.phase.ReadValue());
+            touch.pressure = control.pressure.ReadValue();
+            touch.maximumPossiblePressure = 1.0f;
+            touch.type = control.indirectTouch.IsPressed() ? TouchType.Indirect : TouchType.Direct;
+            touch.altitudeAngle = 0;  // TODO: Not available in TouchControl
+            touch.azimuthAngle = 0;   // TODO: Not available in TouchControl
+
+            var(radius, variance) = RadiusValues(control.radius.ReadValue());
+            touch.radius = radius;
+            touch.radiusVariance = variance;
+
+            return touch;
         }
 
         public static Touch GetTouch(int index)
         {
-            return default;
+            if (index >= touchCount)
+                throw new IndexOutOfRangeException();
```

Comment:
`IndexOutOfRangeException` should not be thrown this way. Use `ArgumentOutOfRangeException` instead.

Reply:
(also need to make sure we are throwing a type equivalent to what the old implementation throws; can be subtype, though)

Reply:
Sorry I mean: throw new ArgumentOutOfRangeException("Invalid touch index.");

Reply:
First arg must be parameter name.

```
throw new ArgumentOutOfRangeException(nameof(index));
```